src/main/java/reposense/util/TimeUtil.java

```diff
@@ -43,4 +48,22 @@ public static String getElapsedTime() {
     public static String getElapsedTimeMessage() {
         return "Elapsed processing time:" + getElapsedTime();
     }
+
+    /**
+     * Get the raw offset in milliseconds for the {@code zoneId} timezone compared to UTC.
+     */
+    public static int getZoneRawOffset(ZoneId zoneId) {
+        Instant now = Instant.now();
+        ZoneOffset zoneOffset = zoneId.getRules().getOffset(now);
+        return zoneOffset.getTotalSeconds() * 1000;
+    }
+
+    /**
+     * Get the {@code current} date that is in {@code fromZoneId} timezone into {@code toZoneId} timezone.
+     */
+    public static Date convertDateWithZones(Date current, ZoneId fromZoneId, ZoneId toZoneId) {
```

Comment:
I think we should stick to using `getSystemDateFromZonedDate`? Right now the `fromZoneId` will always be the system default, which was the original implementation. I don't think we will have a case where the `Date` object is not in the system time zone.